packages/components/filters/filter-menu/src/filter-menu.tsx

```diff
@@ -0,0 +1,109 @@
+import { type ReactElement } from 'react';
+import DropdownMenu from '@commercetools-uikit/dropdown-menu';
+import { Footer } from './footer';
+import { Header } from './header';
+import { TriggerButton } from './trigger-button';
+
+export type TAppliedFilterValue = {
+  key?: string;
+  label: string | ReactElement;
```

Comment:
If you use `ReactNode` you cover everything that react can render and can ditch the `string |`. Additionally, if a consumer needs to render a number, there is no need to transform it into a string first to do so.

Reply:
`ReactNode` works for me

Reply:
fixed in [59a151d](https://github.com/commercetools/ui-kit/pull/2905/commits/59a151d2939c54b37b9a36acc3c70f09fd6fdfde)

Reply:
> this also allows null and undefined on what we are considering to be a required prop. How does TS resolve that? (I still feel like a TS newbie sometimes)

@Sarah4VT I was curious and tried it out:
![image](https://github.com/user-attachments/assets/21e3d5a8-99da-449e-b2fc-5ec6953b53c9)

Apparently you'd still be required to explicitly set the label property to `undefined` etc.

Reply:
Ah that makes sense.